### PR TITLE
Add CDN 2022 05 01 preview version

### PR DESCRIPTION
**Pull request by gubalasu**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by mentat9** [comment]
>           "additionalProperties": {

ARM recommends against the use of the additionalProperties/empty/unschematized object pattern. ARM favors stronger typing wherever possible because it allows all consumers of your API to reason about it. This has benefits for everyone that interacts with your API including basic infrastructure, other Azure features, SDK owners, and customers. Unschematized objects are essentially API contracts that bypass Azure versioning infrastructure.

For example, customers will not be able to manage these objects using Azure Policy, or query them using ARG, both of which are standard Azure features. Customers expect to be able to use all common Azure features against all of their resources. 

I see your RP (and this PR) already uses this pattern a lot, so you might prefer not to change this now. However, we want RPs to move away from it over time except when absolutely needed.

Generally, this pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined).

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/afdx.json:4171 in 52959df. [](commit_id = 52959df68a47f557688da1e7d54802c52e4d5b22, deletion_comment = False)

**Comment by mentat9** [comment]
>         "migrationWafMappings": {

Recommend avoiding acronyms or abbreviations in property names, unless they are very common and standard (e.g. IP or Url): [https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#171-approach](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#171-approach)

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/cdn.json:2817 in 52959df. [](commit_id = 52959df68a47f557688da1e7d54802c52e4d5b22, deletion_comment = False)

**Comment by mentat9** [comment]
>       "description": "Result for canMigrate operation.",

... migrate operation.

---
Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/cdn.json:2845 in 52959df. [](commit_id = 52959df68a47f557688da1e7d54802c52e4d5b22, deletion_comment = False)

File: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/cdnwebapplicationfirewall.json
Comment (mentat9) [inline specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/cdnwebapplicationfirewall.json:408]: >additionalProperties [](http://example.com/codeflow?start=11&length=20)

Avoid this pattern if possible.

**Comment by gubalasu** [comment]
> > ```
> >       "additionalProperties": {
> > ```
> 
> ARM recommends against the use of the additionalProperties/empty/unschematized object pattern. ARM favors stronger typing wherever possible because it allows all consumers of your API to reason about it. This has benefits for everyone that interacts with your API including basic infrastructure, other Azure features, SDK owners, and customers. Unschematized objects are essentially API contracts that bypass Azure versioning infrastructure.
> 
> For example, customers will not be able to manage these objects using Azure Policy, or query them using ARG, both of which are standard Azure features. Customers expect to be able to use all common Azure features against all of their resources.
> 
> I see your RP (and this PR) already uses this pattern a lot, so you might prefer not to change this now. However, we want RPs to move away from it over time except when absolutely needed.
> 
> Generally, this pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined.
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/afdx.json:4171 in [52959df](https://github.com/Azure/azure-rest-api-specs/commit/52959df68a47f557688da1e7d54802c52e4d5b22). [](commit_id = [52959df](https://github.com/Azure/azure-rest-api-specs/commit/52959df68a47f557688da1e7d54802c52e4d5b22), deletion_comment = False)

As you mentioned. We will keep it this way now. We will avoid doing this when we are designing features in the future

**Comment by gubalasu** [comment]
> > ```
> >   "description": "Result for canMigrate operation.",
> > ```
> 
> ... migrate operation.
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/cdn.json:2845 in [52959df](https://github.com/Azure/azure-rest-api-specs/commit/52959df68a47f557688da1e7d54802c52e4d5b22). [](commit_id = [52959df](https://github.com/Azure/azure-rest-api-specs/commit/52959df68a47f557688da1e7d54802c52e4d5b22), deletion_comment = False)

Fixed

**Comment by gubalasu** [comment]
> > ```
> >     "migratedProfileArmResourceId": {
> > ```
> 
> migratedProfileResourceId? names shouldn't include 'Arm'. If you want a modifier for resourceId, consider Azure, as in migratedProfileAzureResourceId.
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/cdn.json:2847 in [52959df](https://github.com/Azure/azure-rest-api-specs/commit/52959df68a47f557688da1e7d54802c52e4d5b22). [](commit_id = [52959df](https://github.com/Azure/azure-rest-api-specs/commit/52959df68a47f557688da1e7d54802c52e4d5b22), deletion_comment = False)

Fixed

**Comment by gubalasu** [comment]
> > ```
> >     "migrationWafMappings": {
> > ```
> 
> Recommend avoiding acronyms or abbreviations in property names, unless they are very common and standard (e.g. IP or Url): https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#171-approach
> 
> Refers to: specification/cdn/resource-manager/Microsoft.Cdn/preview/2022-05-01-preview/cdn.json:2817 in [52959df](https://github.com/Azure/azure-rest-api-specs/commit/52959df68a47f557688da1e7d54802c52e4d5b22). [](commit_id = [52959df](https://github.com/Azure/azure-rest-api-specs/commit/52959df68a47f557688da1e7d54802c52e4d5b22), deletion_comment = False)

Fixed

**Comment by jianyexi** [comment]
@gubalasu could you take look at the 'Swagger Avocado' and 'API Readiness Check' error

**Comment by gubalasu** [comment]
@jianyexi Can you take a look at this PR now?

**Comment by gubalasu** [comment]
@jianyexi Can you take a look at this PR

**Comment by jianyexi** [comment]
@qiaozha  @ArcturusZhang @msyyc could you help confirm if the SDK breaking change needs to review ?

**Comment by ArcturusZhang** [comment]
> @qiaozha @ArcturusZhang @msyyc could you help confirm if the SDK breaking change needs to review ?

I could only approve the SDK breaking change if Jeffrey approves the breaking changes in the swagger. Therefore let us wait for Jeffrey's response first

**Comment by jianyexi** [comment]
> > @qiaozha @ArcturusZhang @msyyc could you help confirm if the SDK breaking change needs to review ?
> 
> I could only approve the SDK breaking change if Jeffrey approves the breaking changes in the swagger. Therefore let us wait for Jeffrey's response first

There is no breaking change in swagger, I removed the 'BreakingChangeReviewRequired' label,  feel free to add it back if you think there are SDK breaking changes

**Comment by gubalasu** [comment]
@jianyexi Can you help merge this

**Review by jianyexi — APPROVED** [review]

**Comment by jianyexi** [comment]
@msyyc could you assist with the python-track2 SDK failure, as it's a required task and will block the PR merge

**Comment by msyyc** [comment]
python failure is fixed

File: specification/cdn/resource-manager/readme.md
Comment (cataggar) [inline specification/cdn/resource-manager/readme.md:31]: @jianyexi, do you know what `tag` means here? Is it the default tag that should be used? https://github.com/Azure/azure-rest-api-specs/issues/20652

**Pull request merged**
